common/src/main/java/com/wynntils/core/chat/tabs/ChatTabModel.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.core.chat.tabs;
+
+import com.wynntils.core.managers.Model;
+import com.wynntils.mc.event.ChatPacketReceivedEvent;
+import com.wynntils.mc.utils.McUtils;
+import com.wynntils.wynn.event.WorldStateEvent;
+import com.wynntils.wynn.model.WorldStateManager;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import net.minecraft.client.gui.components.ChatComponent;
+import net.minecraft.network.chat.Component;
+import net.minecraftforge.eventbus.api.EventPriority;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class ChatTabModel extends Model {
+    private static final ChatComponent EMPTY_CHAT_COMPONENT = new ChatComponent(McUtils.mc());
+
+    private static ChatTab focusedTab = null;
+
+    private static Map<ChatTab, ChatComponent> chatTabData = Map.of();
+
+    public static void init() {
+        chatTabData = new ConcurrentHashMap<>();
+    }
+
+    public static void disable() {
+        chatTabData.clear();
+    }
+
+    @SubscribeEvent
+    public static void onWorldStateChange(WorldStateEvent event) {
+        if (event.getNewState() == WorldStateManager.State.NOT_CONNECTED) {
```

Comment:
I just realized: how well does this code play with logging in to different servers and local worlds? So if you log in to e.g. server A, then Wynncraft, then server B, does everything look fine and dandy, for all three servers?

Reply:
Yes, everything should work fine. Chat tabs are only injected on Wynn.

Reply:
Just to be clear: have you tested, or are you just asserting that "it should work"? :)

Reply:
> Just to be clear: have you tested, or are you just asserting that "it should work"? :)

Tested on Hypixel